client/allocrunner/alloc_runner.go

```diff
@@ -450,6 +446,9 @@ func (ar *allocRunner) Restore() error {
 			return err
 		}
 		states[tr.Task().Name] = tr.TaskState()
+
+		// restore process wrangler for task
+		ar.wranglers.Setup(proclib.Task{AllocID: tr.Alloc().ID, Task: tr.Task().Name})
```

Comment:
Well the `ar.wranglers` field belongs to the alloc runner which the individual task runner wouldn't have access to. Which, I think is technically unnecessary for the current implementation but in the long term it could also manage child processes associated with the alloc (and not any task).